### PR TITLE
[L13] Not using the SafeMath library

### PR DESCRIPTION
**Pull request by ewilz**
### Description

Add safemath where possible

### Other changes

Added `ICeloVersionedContract` interface to `UsingPrecompiles`

### Tested

N/A should remain working as expected

### Related issues

- Fixes https://github.com/celo-org/celo-labs/issues/603
### Backwards compatibility

N/A should remain working as expected

### CODE REVIEW

File: packages/protocol/specs/harnesses/LockedGoldHarness.sol [hunk 1]
```diff
@@ -1,10 +1,13 @@
 pragma solidity ^0.5.8;
 
 import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
```
Comment (yorhodes) [inline packages/protocol/specs/harnesses/LockedGoldHarness.sol:4]: these changes are probably harmless but to clarify, these specs are only used in the `certora-test` formal verification and the contracts are never deployed on network afaik

**Review by yorhodes — APPROVED** [review]
LGTM

File: packages/protocol/specs/harnesses/LockedGoldHarness.sol (hunk 1 above)
Comment (ewilz) [reply packages/protocol/specs/harnesses/LockedGoldHarness.sol:4]: thanks, useful information

File: packages/protocol/specs/harnesses/RegistryHarness.sol [hunk 2]
```diff
@@ -123,23 +126,23 @@ contract RegistryHarness is IRegistryExtended {
...
-    return randomBoolMap[randomIndex];
+    randomIndex.add(1);
+    return randomBoolMap[randomIndex.add(1)
   }
 
   function getRandomUInt256() public returns (uint256) {
-    randomIndex++;
-    return randomUInt256Map[randomIndex];
+    randomIndex.add(1);
+    return randomUInt256Map[randomIndex.add(1)
```
Comment (ewilz) [inline packages/protocol/specs/harnesses/RegistryHarness.sol]: oops

**Pull request merged**